### ISSUE TITLE
metadata.cover not extracting cover art from FLAC metadata

### ISSUE THREAD

**Issue by DigiBC** [savonet/liquidsoap#2950]
**Describe the bug**

The <code>metadata.cover</code> operator works well when extracting cover art (JPEG / PNG) from MP3 metadata, but it seems to fail with FLAC metadata. The logs show the following:

_MP3_   :heavy_check_mark:
```
xxxx/xx/xx xx:xx:xx [metadata.cover:5] APIC not found for /mnt/files/audio_test_2/track_01.mp3, trying builtin tag reader.
xxxx/xx/xx xx:xx:xx [metadata.cover:5] Found APIC for /mnt/files/audio_test_2/track_01.mp3. 
```

_FLAC_   :x:
```
xxxx/xx/xx xx:xx:xx [metadata.cover:5] APIC not found for /mnt/files/audio_test_1/track_01.flac, trying builtin tag reader.
xxxx/xx/xx xx:xx:xx [lang:4] No cover found for /mnt/files/audio_test_1/track_01.flac.
```
The issue seems to be unrelated to the decoder / metadata decoder used.

**To Reproduce**

I use the following code to extract cover art and to save it as JPEG files:
```
def extract_cover_art(m)
  file.write(append=false, data="#{metadata.cover(coverart_mime='image/jpeg', m)}", "/mnt/files/pics/cover_art.jpg")
end
music.on_track(extract_cover_art)
```
The resulting JPEG files from MP3 metadata are valid, but from FLAC metadata only 4 byte files containing _null_ are created.

**Expected behavior**

The <code>metadata.cover</code> operator was designed for the formats MP3, OGG and FLAC.

**Version details**

- OS: Linux Mint 21.1 (Ubuntu 22.04) amd64
- Version: Liquidsoap ≤ 2.1.4

**Install method**

opam

**Comment by smimram** [comment]
Could you please provide a test flac file with coverart which exhibits this? (possibly privately if you want)

**Comment by DigiBC** [comment]
I've prepared two short FLAC files with embedded cover art (PNG and JPEG):

[FLAC_Files_with_Cover_Art.zip](https://github.com/savonet/liquidsoap/files/10933598/FLAC_Files_with_Cover_Art.zip)

Meanwhile I've noticed that you're working on a solution already...

**Comment by smimram** [comment]
This should be fixed by #2952.
